aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,609 @@
+            to_export, filename=full_export_path, overwrite=overwrite, include_comments=True
+        )  # extras are automatically included
+
+    return _export_cache
+
+
+# Do we always want to use hash_code_by_entrypoint here?
+@pytest.fixture(scope='function')
+def load_cache(hash_code_by_entrypoint, absolute_archive_path, import_with_migrate):
+    """Fixture to load a cached AiiDA graph"""
+
+    def _load_cache(path_to_cache=None, node=None):
+        """
+        Function to import an AiiDA graph
+
+        :param path_to_cache: str or path to the AiiDA export file to load,
+                              if path_to_cache points to a directory,
+                              all import files in this dir are imported
+        :param node: AiiDA node, determines which cache to load (CURRENTLY NOT IMPLEMENTED)
+        :raises : OSError or FileNotFoundError, if import file is non existent
+        """
+        if path_to_cache is None:  #pylint: disable=no-else-raise
+            if node is None:
+                raise ValueError(
+                    "Node argument can not be None "
+                    "if no explicit 'path_to_cache' is specified"
+                )
+            raise NotImplementedError('The node argument is currently not supported')
+            #else:  # create path from node
+            #    pass
+            #    # get default data dir
+            #    # get hash for give node
+            #    # construct path from that
+        else:
+            # relative paths given will be completed with cwd
+            full_import_path = absolute_archive_path(path_to_cache)
+
+        if os.path.exists(full_import_path):
+            if os.path.isfile(full_import_path):
+                # import cache, also import extras
+                import_with_migrate(full_import_path)
+            elif os.path.isdir(full_import_path):
+                for filename in os.listdir(full_import_path):
+                    file_full_import_path = os.path.join(full_import_path, filename)
+                    # we curretly assume all files are valid aiida exports...
+                    # maybe check if valid aiida export, or catch exception
+                    import_with_migrate(file_full_import_path)
+            else:  # Should never get there
+                raise OSError(
+                    f"Path: {full_import_path} to be imported exists, but is neither a file or directory."
+                )
+        else:
+            raise FileNotFoundError(f"File: {full_import_path} to be imported does not exist.")
+
+        # need to rehash after import, otherwise cashing does not work
+        # for this we rehash all process nodes
+        # this way we use the full caching mechanism of aiida-core.
+        # currently this should only cache CalcJobNodes
+        _rehash_processes()
+
+    return _load_cache
+
+
+@pytest.fixture(scope='function')
+def with_export_cache(export_cache, load_cache, absolute_archive_path):
```

Comment:
I decided to go with the archive names, since I think it is more intuitive for people getting started if they use the same terminology as the current AiiDA versions.

I took care of most of the suggestions you gave but I want to clean up the commit history a bit to make it easier to review, since the changes are larger
for the moment this is located here https://github.com/JuDFTteam/aiida-testing/tree/archive_cache and I will clean up the branch after the weekend